sdcm/cluster.py

```diff
@@ -3771,35 +3774,22 @@ def critical_node_setup_events():
             add_severity_limit_rules([f'{e["rule_pattern"]}={e["default_severity"]}' for e in critical_events])
 
         start_time = time.perf_counter()
-        init_nodes = []
-        results = []
 
         if isinstance(cl_inst, BaseScyllaCluster):
             # Update installed scylla before node setup, scylla server will be start at the end of node_setup
             cl_inst.update_db_binary(node_list, start_service=False)
             cl_inst.update_db_packages(node_list, start_service=False)
 
         with critical_node_setup_events():
+            start_time = time.perf_counter()
+            cl_inst.run_func_parallel(func=node_setup, node_list=node_list)
             for node in node_list:
-                if isinstance(cl_inst, BaseScyllaCluster) \
-                        and not getattr(cl_inst, 'params', {}).get('use_legacy_cluster_init'):
-                    init_nodes.append(node)
-                    start_time = time.perf_counter()
-                    node_setup(node)
-                    verify_node_setup(start_time)
-                else:
-                    setup_thread = threading.Thread(target=node_setup, name='NodeSetupThread',
-                                                    args=(node,), daemon=True)
-                    setup_thread.start()
-                    if isinstance(cl_inst, BaseScyllaCluster):
```

Comment:
I'm not sure why https://github.com/scylladb/scylla-cluster-tests/commit/74bf8799ba6a3f4ee946dd3ed780b0c7f3279ef5 
add those 2 min sleep between nodes 

@aleksbykov do you remember what was the reason for that ?

Reply:
it was related, that node could not bootstrap in parallel on elder versions. they need seed nodes to be loaded